Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListRowWithValidationFailureComparerTest.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Moq;
+using NUnit.Framework;
+using Remotion.ObjectBinding.Validation;
+using Remotion.ObjectBinding.Web.UI.Controls;
+using Remotion.ObjectBinding.Web.UI.Controls.Validation;
+
+namespace Remotion.ObjectBinding.Web.UnitTests.UI.Controls
+{
+  [TestFixture]
+  public class BocListRowWithValidationFailureComparerTest
+  {
+    private BocListRowWithValidationFailureComparer _comparer;
+    private IBocListValidationFailureRepository _validationFailureRepository;
+
+    private IBusinessObject _zeroErrorBusinessObject;
+    private IBusinessObject _singleErrorBusinessObject;
+    private IBusinessObject _doubleErrorBusinessObject;
+
+    private readonly BocListRow _nullRow = null;
+    private BocListRow _zeroErrorRow;
+    private BocListRow _singleErrorRow;
+    private BocListRow _doubleErrorRow;
+
+    [SetUp]
+    public void Setup ()
+    {
+      _validationFailureRepository = new BocListValidationFailureRepository();
+      _comparer = new BocListRowWithValidationFailureComparer(_validationFailureRepository);
+
+      _singleErrorBusinessObject = new Mock<IBusinessObject>().Object;
+      _zeroErrorBusinessObject = new Mock<IBusinessObject>().Object;
+      _doubleErrorBusinessObject = new Mock<IBusinessObject>().Object;
+
+      _zeroErrorRow = new BocListRow(0, _zeroErrorBusinessObject);
+      _singleErrorRow = new BocListRow(1, _singleErrorBusinessObject);
+      _doubleErrorRow = new BocListRow(2, _doubleErrorBusinessObject);
+
+      _validationFailureRepository.AddValidationFailuresForDataRow(_zeroErrorBusinessObject, new BusinessObjectValidationFailure[0]);
+      _validationFailureRepository.AddValidationFailuresForDataRow(_singleErrorBusinessObject, new[] { BusinessObjectValidationFailure.Create("Single error") });
+      _validationFailureRepository.AddValidationFailuresForDataRow(
+          _doubleErrorBusinessObject,
+          new[]
+          {
+            BusinessObjectValidationFailure.Create("Double error 1"),
+            BusinessObjectValidationFailure.Create("Double error 2")
+          });
+    }
+
+    [Test]
+    public void Compare_WithFirstObjectNull_ReturnsMinusOne ()
+    {
+      Assert.That(_comparer.Compare(_nullRow, _singleErrorRow), Is.EqualTo(1));
+    }
+
+    [Test]
+    public void Compare_WithSecondObjectNull_ReturnsOne ()
+    {
+      Assert.That(_comparer.Compare(_singleErrorRow, _nullRow), Is.EqualTo(-1));
+    }
```

Comment:
Naming:
`_ReturnsNegativeOne`
`_ReturnsPositiveOne`
The actual `One` might be redudnant, `_ReturnsPositiveNumber` etc would be sufficient.